actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -108,15 +110,17 @@ class ActionCable.Connection
       ActionCable.log("WebSocket onclose event")
       if not @isProtocolSupported()
         ActionCable.log("WebSocket is closing because there is no common protocol. Stopping ConnectionMonitor.")
-        @monitor.stop()   
-      @disconnect()
+        @disconnect(ActionCable.Events.Disconnected.ProtocolNotSupported)
+      else
+        @disconnect(new ActionCable.Events.Disconnected(event.code, event.reason, true))
```

Comment:
Subscriptions don't receive Event objects anywhere else so I think passing a plain object would do. Perhaps:
```js
 `{ code: 123, message: "...", willReconnect: (true|false) }
```
I think `willReconnect` better indicates wether action is needed.